src/catalog/catalog_entry/duck_table_entry.cpp

```diff
@@ -364,7 +420,14 @@ unique_ptr<CatalogEntry> DuckTableEntry::AddColumn(ClientContext &context, AddCo
 
 	vector<unique_ptr<Expression>> bound_defaults;
 	auto bound_create_info = binder->BindCreateTableInfo(std::move(create_info), schema, bound_defaults);
-	auto new_storage = make_shared_ptr<DataTable>(context, *storage, info.new_column, *bound_defaults.back());
+
+	// Store the expression in AlterInfo to keep it alive (the executor only stores a pointer)
+	info.default_expression = std::move(bound_defaults.back());
+
+	info.default_executor.reset(new ExpressionExecutor(context));
```

Comment:
As I've said before, using `new` is almost always a bad idea, I don't think this is an exception
`info.default_executor = make_uniq<ExpressionExecutor>(context);` is what you're looking for I think.